File: scrolltotop/src/main/java/com/mahdikh/vision/scrolltotop/animator/FadeAnimator.java
package com.mahdikh.vision.scrolltotop.animator;

import android.view.View;

public class FadeAnimator extends BaseAnimator {
    private float maxAlpha;

    public FadeAnimator(float maxAlpha) {
        this.maxAlpha = maxAlpha;
    }

    public FadeAnimator() {
        maxAlpha = 0.8F;
    }

    @Override
    public void onPrepare(View view) {
        view.setAlpha(0.0F);
        view.setVisibility(View.GONE);
    }

    @Override
    public void onShow(View view) {
        view.setVisibility(View.VISIBLE);
        view.setAlpha(0.0F);
        view.animate()
                .alpha(maxAlpha)
                .setDuration(getDuration())
                .setInterpolator(getInterpolator());
    }

    @Override
    public void onHide(View view) {
        view.animate()
                .alpha(0F)
                .setDuration(getDuration())
                .setInterpolator(getInterpolator())
                .withEndAction(() -> view.setVisibility(View.GONE));
    }

    public FadeAnimator setMaxAlpha(float maxAlpha) {
        this.maxAlpha = maxAlpha;
        return this;
    }

    public float getMaxAlpha() {
        return maxAlpha;
    }
}

File: scrolltotop/src/main/java/com/mahdikh/vision/scrolltotop/animator/FadeAnimator.kt
package com.mahdikh.vision.scrolltotop.animator

import android.view.View

open class FadeAnimator : BaseAnimator {
    var maxAlpha: Float

    constructor() : super() {
        maxAlpha = 0.8F
    }

    constructor(maxAlpha: Float) : super() {
        this.maxAlpha = maxAlpha
    }

    override fun prepare(view: View) {
        view.visibility = View.GONE
        view.alpha = 0.0F
    }

    override fun preAnimateShow(view: View) {
        view.visibility = View.VISIBLE
        view.alpha = 0.0F
    }

    override fun animateShowImpl(view: View) {
        view.animate().apply {
            alpha(maxAlpha)
            duration = this@FadeAnimator.duration
            interpolator = null
        }.start()
    }

    override fun animateHideImpl(view: View) {
        view.animate().apply {
            alpha(0.0F)
            duration = this@FadeAnimator.duration
            interpolator = null
            withEndAction { view.visibility = View.GONE }
        }.start()
    }
}